pkg/endpoint/policy.go

```diff
@@ -1114,30 +1094,24 @@ func (e *Endpoint) runIPIdentitySync(endpointIP addressing.CiliumIP) {
 					e.Mutex.RUnlock()
 					return nil
 				}
-				//identityValue := e.SecurityIdentity.ID.StringID()
-				ipIDPair := identityPkg.IPIdentityPair{
-					IP:       endpointIP.IP(),
-					ID:       e.SecurityIdentity.ID,
-					Metadata: e.FormatGlobalEndpointID(),
-				}
+
+				IP := endpointIP.IP()
+				ID := e.SecurityIdentity.ID
+				metadata := e.FormatGlobalEndpointID()
 
 				// Release lock as we do not want to have long-lasting key-value
 				// store operations resulting in lock being held for a long time.
 				e.Mutex.RUnlock()
 
-				marshaledIPIDPair, err := json.Marshal(ipIDPair)
-				if err != nil {
-					return err
-				}
-
-				if err := kvstore.Update(ipKey, marshaledIPIDPair, true); err != nil {
-					return fmt.Errorf("unable to add endpoint IP '%s' to identity '%s': %s", ipKey, marshaledIPIDPair, err)
+				if err := ipcache.UpsertIPToKVStore(IP, ID, metadata); err != nil {
+					return fmt.Errorf("unable to add endpoint IP mapping '%s'->'%d': %s", IP.String(), ID, err)
 				}
 				return nil
 			},
 			StopFunc: func() error {
-				if err := kvstore.Delete(ipKey); err != nil {
-					return fmt.Errorf("unable to delete endpoint IP '%s': %s", ipKey, err)
+				ip := endpointIP.String()
+				if err := ipcache.DeleteIP(ip); err != nil {
```

Comment:
Can you make `DeleteIP` mirror `UpsertIPToKVStore` so that it is clear that it is being deleted from the key-value store, and not the ipcache itself directly?